### ISSUE TITLE
run SegNAFNet_arch

### ISSUE THREAD

**Issue by Nanncy99** [HPLQAQ/SAM-Deblur#7]
![image](https://github.com/HPLQAQ/SAM-Deblur/assets/74661436/3ee1c446-3016-4ba4-a30e-681267215afe)
What should I do if the forword function lacks masks when calculating model parameters?Hope to get your answer, thank you very much!

**Comment by HPLQAQ** [comment]
Hello, our method needs to get masks calculated from SAM model in advance.
To run the codes, the dataset that we provided is needed.


Download prepared data for experiments from [Baidu Netdisk](https://pan.baidu.com/s/1jHgwKqFuyyy8yixByiEfCg?pwd=hplv)|[Onedrive](https://1drv.ms/f/s!Ar-cR-XYHDTtgt8pYy2MBqQvroXl-Q?e=U7o4xc).  
If you only want to run our deblur pipeline, download test/val datasets(GoPro, RealBlurJ, REDS, ReLoBlur provided). if you want to train the model yourself, download train datasets(GoPro provided). Check [datasets README](datasets/README.md) for standard dataset structure.
Unzip data and put under **datasets** dir for experiments.


The mask generation codes are in
```
scripts/data_preparation/1_create_masks_with_sam.py: Create masks for concat method.
scripts/data_preparation/2_masks_to_grouped_masks.py: Create grouped_masks from masks for our MAP method.
```
I'll try integrating sam into the inference code later to avoid the necessity of downloading our data when I'm available.


- masks (torch.Tensor): A tensor of shape (M, 1, H, W) representing M masks.
You can get the masks following the mask generation methods decribed in Zheyan Jin, Shiqi Chen, Yueting Chen, Zhihai Xu, and
Huajun Feng, “Let segment anything help image dehaze,”
arXiv preprint arXiv:2306.15870, 2023. or have a look at our codes in scripts/data_preparation/1_create_masks_with_sam.py.
**Only to get the code running, you can pass torch.ones(1, 1, H, W) to the function.**

**Comment by Nanncy99** [comment]
Thank you very much for your answer, it has been very useful to me.However, I have one more small question.Where is MAP Unit reflected in the code in the paper? I don't seem to have found it yet.I hope to receive your reply. Thank you very much again.

**Comment by HPLQAQ** [comment]
https://github.com/HPLQAQ/SAM-Deblur/blob/9e0c8cf20f4e00915d72415a85416795417001bd/basicsr/models/archs/SegNAFNet_arch.py#L104

From line 104 to line 110, we calculated the area and the sum_val for each mask to get  the avg_val.

From line 112 to line 115, they were multiplied to the masks and added together.

if you remove the sum operation at line 115, then from line 104 to 115 reflects a MAP Unit.

**Comment by Nanncy99** [comment]
Thank you for your answer.I have a small question regarding `scripts/data_preparation/2_masks_to_grouped_masks.py: Create grouped_masks from masks for our MAP method.`
The code uses an lmdb format dataset. If my dataset does not have an lmdb format, do I need to generate an lmdb format?Hope to get your answer, thank you very much!

**Comment by HPLQAQ** [comment]
You can directly change the 'read_image_from_lmdb' function to 'read_image_from_folder' function.

Change the following codes to simply go through the data folder and pass the data path to 'read_image_from_folder' should work.
```
env = lmdb.open(lmdb_dir, readonly=True, max_dbs=0)

with env.begin() as txn:
    cursor = txn.cursor()
    for key, _ in cursor:
```
